src/board.py

```diff
@@ -138,18 +118,20 @@ def __getitem__(self, square: Square | Indices | None) -> Piece | None:
             The piece on the given square.
         """
         if square:
-            i, j = self._indices(square) if isinstance(square, Square) else square
-            return self._board[i][j]
+            if isinstance(square, str):
+                square = Square(square)
+            return self._board[square.rank][square.file]
 
-    def __delitem__(self, square: Square | Indices | None):
+    def __delitem__(self, square: Square | str | None):
         """Remove the piece of a given square.
 
         Args:
             square: The rank and file of the square on which to remove a piece (if any).
         """
         if square:
-            i, j = self._indices(square) if isinstance(square, Square) else square
-            self._board[i][j] = None
+            if isinstance(square, str):
```

Comment:
No usage of `_square` here? If not, delete method.
Alternatively, do `square = Square(square)` even when `square` is a `Square`, it should work.